pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -124,36 +127,51 @@ func (ds *dockerService) StopPodSandbox(podSandboxID string) error {
 		namespace = m.Namespace
 		name = m.Name
 	} else {
-		checkpoint, err := ds.checkpointHandler.GetCheckpoint(podSandboxID)
-		if err != nil {
-			glog.Errorf("Failed to get checkpoint for sandbox %q: %v", podSandboxID, err)
-			return fmt.Errorf("failed to get sandbox status: %v", statusErr)
-		}
+		var checkpoint *PodSandboxCheckpoint
+		checkpoint, checkpointErr = ds.checkpointHandler.GetCheckpoint(podSandboxID)
 		namespace = checkpoint.Namespace
 		name = checkpoint.Name
 		// Always trigger network plugin to tear down
 		needNetworkTearDown = true
 	}
 
+	// Proceed if both sandbox container and checkpoint could not be found. This means that following
+	// actions will only have sandbox ID and not have pod namespace and name information.
+	// Return error if encounter any unexpected error.
+	if checkpointErr != nil {
```

Comment:
Although there will be more indent, I still think it would be more clear to move this block into previous `else`.